### PR TITLE
Web Socket Setup & Notifications Page Creation

### PR DESCRIPTION
**Pull request by bryanansong**
## Description
In this pull request, I set up the connection between the frontend and the backend using the socket.io and also created the notifications page along with the notifications list. I have attached an image of this page with some random sample data just for demonstration. 
<what this pull request is doing>
<what will be done in later PRs and not included here>

## Test Plan
Screenshot 2024-07-20 at 8 50 15 PM

<insert images or gifs of feature>
<otherwise, say how code was tested if not UI facing>
<any edge cases you might have specifically tested>

### CODE REVIEW

File: frontend/src/components/NotificationsList/NotificationsList.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        <p>No notifications</p>
+      ) : (
+        <ul className={styles.notificationsList}>
+          {notifications.map((notification) => (
+            <li key={notification.id} className={styles.notificationItem}>
+              <p>{notification.content}</p>
+              <small>{new Date(notification.createdAt).toLocaleString()}</small>
+              {notification.status !== "read" && (
+                <button onClick={() => handleMarkAsRead(notification.id)}>
+                  Mark as read
```
Comment (Wang96) [inline frontend/src/components/NotificationsList/NotificationsList.tsx:50]: I think we shouldn't let users mark notifications as read. Especially if users come back after a long time and they might have a long list of notifications and it would be quite burdensome for them to mark every single notif as read. As long as the notif is rendered and is viewable to users, we should automatically mark them as read.

Comment (bryanansong) [reply frontend/src/components/NotificationsList/NotificationsList.tsx:50]: I got it, I will tackle this when I am working on stretch features.

File: backend/notifications/notificationManager.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+const selectAndSendNotifications = async () => {
+  const users = await prisma.user.findMany();
+
+  for (const user of users) {
+    const recentNotifications = await prisma.notification.findMany({
+      where: {
+        userId: user.id,
+        createdAt: { gte: new Date(Date.now() - 24 * 60 * 60 * 1000) },
+        sentAt: null,
```
Comment (m5matthew) [inline backend/notifications/notificationManager.js]: Do we need to filter for unread notifs here?
